lib/gauge_polyakov_loop.cu

```diff
@@ -0,0 +1,276 @@
+#include <gauge_field.h>
+#include <gauge_tools.h>
+#include <instantiate.h>
+#include <tunable_nd.h>
+#include <tunable_reduction.h>
+#include <kernels/gauge_polyakov_loop.cuh>
+
+namespace quda {
+
+  template<typename Float, int nColor, QudaReconstructType recon>
+  class GaugeInsertTimeslice : public TunableKernel2D {
+    GaugeField &u;
+    const GaugeField &s;
+    int timeslice;
+    unsigned int minThreads() const { return s.LocalVolumeCB(); }
+
+  public:
+    GaugeInsertTimeslice(GaugeField &u, const GaugeField &s, int timeslice) :
+      TunableKernel2D(u, 2),
+      u(u),
+      s(s),
+      timeslice(timeslice)
+    {
+      strcat(aux, ",4d_vol=");
+      strcat(aux, u.VolString());
```

Comment:
Why is this needed?  The 4-d volume will be included automatically already in the `TuneKey`

Reply:
iirc it's because the `TuneKey` included the number of sites as opposed to the `##x##x##x##` volume?

Reply:
Discussed offline, fixed locally.